pymeasure/experiment/results.py

```diff
@@ -138,12 +149,54 @@ def format(self, record):
         :type record: dict
         :return: a string
         """
-        return self.delimiter.join('{}'.format(record[x]) for x in self.columns)
+        return self.delimiter.join(
+            '{}'.format(record[x]) for x in self.columns)
 
     def format_header(self):
         return self.delimiter.join(self.columns)
 
 
+class CSVFormatter_Pandas(Results_Formatter):
+    """ Formatter of data results, pandas dataframe or single-line CSV """
+
+    def format(self, record):
+        """Formats a record as csv using pandas built-in ``to_csv`` method.
+        Accepts a pandas dataframe or a dict of values matching
+        the given list of columns.
+
+        :param record: record to format.
+        :type record: pandas.DataFrame or dict
+        :return: str
+        """
+        if isinstance(record, pd.DataFrame):
+            pass
```

Comment:
I'd suggest to ensure the correct order and columns are printed using a line like this:
```suggestion
            record = record.reindex(columns=self.columns)
```